### PR TITLE
feat(auto-balance): disk in pressure

### PR DESCRIPTION
**Pull request by c3y1huang**
#### Which issue(s) this PR fixes:

Issue longhorn/longhorn#4105

#### What this PR does / why we need it:

Introduce new fields to `FilesSyncRequest` and `EngineReplicaAddRequest` to handle local file copying when replicas are on the same node.

#### Special notes for your reviewer:

`None`

#### Additional documentation or context

`None`

### CODE REVIEW

File: protobuf/imrpc/proxy.proto [hunk 1]
```diff
@@ -245,6 +245,12 @@ message EngineReplicaAddRequest {
 	int32 file_sync_http_client_timeout = 7;
 	string replica_name = 8;
 	int64 grpc_timeout_seconds = 9;
+	EngineReplicaLocalSync local_sync = 10;
+}
+
+message EngineReplicaLocalSync{
```
Comment (derekbit) [inline protobuf/imrpc/proxy.proto:251]:
```suggestion
message EngineReplicaLocalSync {
```

File: protobuf/ptypes/syncagent.proto [hunk 2]
```diff
@@ -66,6 +66,12 @@ message FilesSyncRequest {
   bool fast_sync = 4;
   int32 file_sync_http_client_timeout = 5;
   int64 grpc_timeout_seconds = 6;
+  FileLocalSync local_sync = 7;
+}
+
+message FileLocalSync{
```
Comment (derekbit) [inline protobuf/ptypes/syncagent.proto:72]:
```suggestion
message FileLocalSync {
```

File: protobuf/imrpc/proxy.proto [hunk 3]
```diff
@@ -245,6 +245,12 @@ message EngineReplicaAddRequest {
...
+	string source = 1;
+	string target = 2;
```
Comment (derekbit) [inline protobuf/imrpc/proxy.proto:253]: nit: suggest using `source_path` and `target_path`

File: protobuf/ptypes/syncagent.proto [hunk 4]
```diff
@@ -66,6 +66,12 @@ message FilesSyncRequest {
...
+	string source = 1;
+	string target = 2;
```
Comment (derekbit) [inline protobuf/ptypes/syncagent.proto:74]: nit: same as above

**Review by derekbit — APPROVED** [review]
LGTM

**Comment by derekbit** [comment]
@c3y1huang Is it ready?

**Comment by c3y1huang** [comment]
> @c3y1huang Is it ready?

Ready for review.

**Review by derekbit — APPROVED** [review]

**Pull request merged**